### ISSUE TITLE
Make PS-DSWP work 

### ISSUE THREAD

**Issue by vgene** [PrincetonUniversity/cpf#27]
Make PS-DSWP whole pipeline work on KS

**Comment by sapostolakis** [comment]
Note that I had the whole pipeline for PS-DSWP working for KS back on June 4th 2019. This might help when trying to get it working again. Back then I reported for KS 2.4x automatic speedup on 8 cores. Spawning workers only once was critical for KS performance since I targeted an inner loop, and functionality-wise handling conditional live-outs and some weird form of reduction were necessary. At some point that summer I changed focus to DOALL and probably some parts (probably codegen and runtime) got broken for PS-DSWP on the way, but it should not be hard to get back on.

**Comment by sapostolakis** [comment]
Note that I started with KS because it did not require any speculation and it was a good first test for the PS-DSWP code gen.
The real challenge for PS-DSWP is to support the validation of all the available forms of speculation. See section 9.2.1 in my thesis (https://liberty.princeton.edu/Publications/phdthesis_sa8.pdf#chapter.9). I have discussed that with Susan and she included some thoughts on her Young Arch workshop talk last year.

**Comment by vgene** [comment]
Susan identified the proble to be reduxremed being turned off and is in the process of getting that back to life.

**Comment by sapostolakis** [comment]
Yes, reg redux need to be re-activated. I disabled it at some point to avoid some hassle with LoopDependenceInfo from NOELLE. But my guess is that after planning works again for KS, codegen and runtime will also need some tweaks to get them working again for KS.
